tool_aws/s3/rm.py

```diff
@@ -360,19 +361,16 @@ def deleteWithPrefix(opts, S3Bucket, keys):
         # Make sure we delete the first batch
         previousNumberOfKeys = keys.maxKeys
         while len(keys) > 0 and previousNumberOfKeys == keys.maxKeys:
-            if pm:
+            if executor:
                 keys = S3Keys(S3Bucket, opts.prefix)
                 keys.chunk(chunkSize)
                 if len(keys):
                     logger.info('New batch delete')
                     logger.info(str(keys))
             if len(keys):
-                pm = PoolManager(numProcs=opts.nbThreads)
-                pm.imap_unordered(
-                    deleteKeys,
-                    keys,
-                    1,
-                    callback=callback)
+                with ProcessPoolExecutor(
+                        max_workers=opts.nbThreads) as executor:
+                    executor.map(deleteKeys, keys, timeout=3 * 60, chunksize=1)
```

Comment:
This won't work, the `executor` outside of the `with ProcessPoolExecutor...` will always be `None`

Reply:
Thanks. I removed the condition